agent/consul/server.go

```diff
@@ -1047,6 +1049,17 @@ func (s *Server) GetWANCoordinate() (*coordinate.Coordinate, error) {
 	return s.serfWAN.GetCoordinate()
 }
 
+func (s *Server) ServerAddr(id raft.ServerID) (raft.ServerAddress, error) {
+	if string(id) == string(s.config.NodeID) {
+		return raft.ServerAddress(s.config.RPCAddr.String()), nil
+	}
+	addr, err := s.router.GetServerAddressByID(s.config.Datacenter, string(id))
```

Comment:
We don't want to use the router for this since it will be using the WAN pool, which is confusing. It will work, but it's not really the right source of information.  Since this isn't called often I think it's fine to just call `Members()` on the `serfLAN` instance and look for a match:

```
for _, member := range s.serfLAN.Members() {
    if ok, s := metadata.IsConsulServer(); ok && s.ID == id {
      return s.Addr.String()
    }
}
return errNotFound
```